cirq/ops/common_gates.py

```diff
@@ -982,3 +982,74 @@ def Ry(rads: float) -> RotYGate:
 def Rz(rads: float) -> RotZGate:
     """Returns a gate with the matrix e^{-i Z rads / 2}."""
     return RotZGate(rads=rads, global_shift_in_half_turns=-0.5)
+
+
+class XXPowGate(eigen_gate.EigenGate,
+             gate_features.TwoQubitGate,
+             gate_features.InterchangeableQubitsGate):
+    """
+    The gate implements the following unitary:
+    X⊗X = [0 0 0 1]
+          [0 0 1 0]
+          [0 1 0 0]
+          [1 0 0 0]
+
+    """
+
+    def __init__(self, *,  # Forces keyword args.
+                 exponent: Optional[Union[value.Symbol, float]] = None) -> None:
+        """
+        Initializes the gate.
+
+        The exponent may be specified.
+        If no argument is given, the default value of
+        exponent = 1 is used.
+
+        Args:
+            exponent: The power that the gate operation will be raised to.
+        """
+
+        super().__init__(exponent=value.chosen_angle_to_half_turns(half_turns=exponent))
+
+    @property
+    def exponent(self) -> Union[value.Symbol, float]:
+        return self._exponent
+
+    def _eigen_components(self):
```

Comment:
Add `def _eigen_shift(self): return (0, 1)`, which will accelerate the period computations.